### ISSUE TITLE
Provide a API to know if `Build_info.version ()` is returning a snapshot version

### ISSUE THREAD

**Issue by toots** [ocaml/dune#4453]
It would be useful to print different messages in the binary to know if we are running a development version or not. Since `Build_info` already knows this information, it would seem like an easy addition to its API.

**Comment by jeremiedimino** [comment]
IIRC, the idea is to explain whether the version number comes from the vcs or was written in the `dune-project` file?

**Comment by toots** [comment]
Yes, we have a specific warning that we print when the application is compiled from a development snapshot.

**Comment by jeremiedimino** [comment]
Ack. Yes, that seems fine to add. Do you want to give a shot a writing a PR for this?

**Comment by toots** [comment]
I'm looking into it! The `version` type is tightly integrated with dune's `Package` and internals. I'm thinking of adding a secondary element, e.g. `type version = string * [ `Dune_project | `Git ]` that could be extended if more cvs are supported and propagate that everywhere.

This might be a bit intrusive so just wanted to check if that's a viable option for you/y'all..

**Comment by rgrinberg** [comment]
> This might be a bit intrusive so just wanted to check if that's a viable option for you/y'all..

Sounds good to me.

> Which version type are you thinking about? Dune also supports mercurial btw.

Yes, so it should be `VCS` rather than `Git`.
